### PR TITLE
Add implementation of widget filtering based on local json configuration

### PR DESCRIPTION
**Pull request by piiq**
Hackity hack mcHackface

### CODE REVIEW

**Comment by montezdesousa** [comment]
I suggest to rename the file to `widgets_config.json` and use this format to easily add other stuff if we need in the future

```json
{
    "exclude": [
        "widget_id_foo",
        "widget_id_bar",
    ]
}
```

**Review by montezdesousa — CHANGES_REQUESTED** [review]
I think we should rename the file to `widgets_settings.json` and use a more general format to easily add other stuff if we need in the future

Example:
```json
{
    "exclude": [
        "widget_id_foo",
        "widget_id_bar",
    ]
}
```

**Comment by deeleeramone** [comment]
Can we also add this as a kwarg of the openbb-api command line script? `--filter`

**Review by deeleeramone — COMMENTED** [review]
Maybe I'm missing something, but how does this new JSON file get created by hub? Wouldn't something need to change in the Hub service in the main repo to know this file was incoming and to create it?

File: openbb_platform/extensions/platform_api/openbb_platform_api/main.py [hunk 1]
```diff
@@ -19,7 +22,20 @@
 CURRENT_USER_SETTINGS = os.path.join(HOME, ".openbb_platform", "user_settings.json")
 USER_SETTINGS_COPY = os.path.join(HOME, ".openbb_platform", "user_settings_backup.json")
 
-FIRST_RUN = True
+# Load widget_exclude_filter.json if it exists
+# Widget filtering is optional and can be used to exclude widgets from the widgets.json file
+# You can generate this filter on OpenBB Hub: https://my.openbb.co/app/platform/widgets
```
Comment (deeleeramone) [inline openbb_platform/extensions/platform_api/openbb_platform_api/main.py:27]: Can I test this? I currently get a 500 error on both `.co` and `.dev`.

Comment (piiq) [reply openbb_platform/extensions/platform_api/openbb_platform_api/main.py:27]: Currently available here https://github.com/OpenBB-finance/openbb-hub/pull/398

**Comment by piiq** [comment]
> Can we also add this as a kwarg of the openbb-api command line script? `--filter`

@deeleeramone I've added a `--no-filter` argument that when provided will trigger "not filtering"

I'll merge this now so that I can deal with the packaging of this extension

**Pull request merged**